test/specs/scale.logarithmic.tests.js

```diff
@@ -690,142 +690,433 @@ describe('Logarithmic Scale tests', function() {
+				firstTick: 1,
+				lastTick: 80,
+				describe: 'empty dataset, ticks.max: 80, without ticks.min'
+			},
+			{
+				dataset: [],
+				scale: {ticks: {max: 0.8}},
+				firstTick: 0.01,
+				lastTick: 0.8,
+				describe: 'empty dataset, ticks.max: 0.8, without ticks.min'
+			},
+			{
+				dataset: [{x: 10, y: 10}, {x: 5, y: 5}, {x: 1, y: 1}, {x: 25, y: 25}, {x: 78, y: 78}],
+				firstTick: 1,
+				lastTick: 80,
+				describe: 'dataset min point {x: 1, y: 1}, max point {x:78, y:78}'
+			},
+		];
+		config.forEach(function(setup) {
+			var axes = [
+				{
+					id: 'x', // horizontal scale
+					start: 'left',
+					end: 'right'
+				},
+				{
+					id: 'y', // vertical scale
+					start: 'bottom',
+					end: 'top'
+				}
+			];
+			axes.forEach(function(axis) {
+				var expectation = 'min = ' + setup.firstTick + ', max = ' + setup.lastTick;
+				describe(setup.describe + ' and axis is "' + axis.id + '"; expect: ' + expectation + ';', function() {
+					beforeEach(function() {
+						var xScaleConfig = {
+							type: 'logarithmic',
+						};
+						var yScaleConfig = {
+							type: 'logarithmic',
+						};
+						var data = setup.data || {
+							datasets: [{
+								data: setup.dataset
+							}],
+						};
+						Chart.helpers.extend(xScaleConfig, setup.scale);
+						Chart.helpers.extend(yScaleConfig, setup.scale);
+						Chart.helpers.extend(data, setup.data || {});
+						this.chart = window.acquireChart({
+							type: 'line',
+							data: data,
+							options: {
+								scales: {
+									xAxes: [xScaleConfig],
+									yAxes: [yScaleConfig]
+								}
+							}
+						});
+					});
+
+					it('should get the correct pixel value for a point', function() {
+						var chart = this.chart;
+						var axisID = axis.id + '-axis-0';
+						var scale = chart.scales[axisID];
+						var firstTick = setup.firstTick;
+						var lastTick = setup.lastTick;
+						var start = chart.chartArea[axis.start];
+						var end = chart.chartArea[axis.end];
+
+						expect(scale.getPixelForValue(firstTick, 0, 0)).toBe(start);
+						expect(scale.getPixelForValue(lastTick, 0, 0)).toBe(end);
+						expect(scale.getPixelForValue(0, 0, 0)).toBe(start); // 0 is invalid, put it at the start.
+
+						expect(scale.getValueForPixel(start)).toBeCloseTo(firstTick, 4);
+						expect(scale.getValueForPixel(end)).toBeCloseTo(lastTick, 4);
+
+						chart.scales[axisID].options.ticks.reverse = true; // Reverse mode
+						chart.update();
+
+						// chartArea might have been resized in update
+						start = chart.chartArea[axis.end];
+						end = chart.chartArea[axis.start];
+
+						expect(scale.getPixelForValue(firstTick, 0, 0)).toBe(start);
+						expect(scale.getPixelForValue(lastTick, 0, 0)).toBe(end);
+
+						expect(scale.getValueForPixel(start)).toBeCloseTo(firstTick, 4);
+						expect(scale.getValueForPixel(end)).toBeCloseTo(lastTick, 4);
+					});
+				});
+			});
+		});
+	});
+
+	describe('when', function() {
+		var config = [
+			{
```

Comment:
would it make sense to test `ticks.min == 0` and `ticks.max == 0` at the same time? or is that covered in another test?

Reply:
The `determineDataLimits` function would generate `{min: 1, max: 10, minNotZero: 1}` values. However the [core.ticks.js#137](https://github.com/chartjs/Chart.js/blob/master/src/core/core.ticks.js#L137) would create `lastTick = 0` in this case causing the ticks to be: `[0,1,2,3,4,5,6,7,8,9,0]`. The same would happen with other invalid input.

The better question would be. Should Chart.js have a checking function for the configuration as it would allow invalid input `ticks: {min: 10, max: 0}` and this would bubble through the `determineDataLimits` and would cause problems in the tick generator?

Reply:
I think it's OK to not handle that case. The user would have had to explicitly set it that way which would not make sense. 

In terms of general strategy, I'm not sure how much checking we want to do here. @simonbrunel @benmccann thoughts?

Reply:
maybe, although the explicitly set option might be the result of some other function. Sanitizing the data and config, could make the code not having to deal with such situations.